tests/pos-macros/i16630.scala

```diff
@@ -0,0 +1,6 @@
+import scala.language.experimental.macros
+
+trait TraitWithTypeParam[A] {
+  protected inline def foo: Option[A] = ${ ??? }
+  protected def foo: Option[A] = macro ???
+}
```

Comment:
I would rather minimize the test as
```scala
import scala.language.experimental.macros

trait TraitWithTypeParam[A] {
  protected inline def foo: Foo[A] = ${ MacrosImpl.fooImpl[A] }
  protected def foo: Foo[A] = macro MacrosImpl.compatFooImpl[A]
}

object MacrosImpl {
  def fooImpl[A: Type](using quotes: Quotes): Expr[Foo[A]] = ???

  def compatFooImpl[A: c.WeakTypeTag](c: scala.reflect.macros.blackbox.Context): c.Tree = ???
```

There are some extra checks on the `macro MacrosImpl.compatFooImpl[A]` that might trigger other issues.

Reply:
Right. You can create a dummy interface to test this. Here is an example https://github.com/lampepfl/dotty/blob/main/tests/pos-macros/i8945.scala#L2-L11

Reply:
Thanks for all the pointers both of you.  I have updated the test roughly as per your suggestions.  Next I will work out how to squash the commits.